classes/ReassuranceActivity.php

```diff
@@ -0,0 +1,175 @@
+*  International Registered Trademark & Property of PrestaShop SA
+*/
+
+class ReassuranceActivity extends ObjectModel
+{
+    public $id;
+    public $icone;
+    public $icone_perso;
+    public $title;
+    public $description;
+    public $status;
+    public $position;
+    public $id_shop;
+    public $type_link;
+    public $link;
+    public $id_cms;
+    public $date_add;
+    public $date_upd;
+
+
+    /**
+     * @see ObjectModel::$definition
+     */
+    public static $definition = array(
+        'table'     => 'psreassurance',
+        'primary'   => 'id_psreassurance',
+        'multilang' => true,
+        'multilang_shop' => true,
+        'fields'    => array(
+            'icone'         => array('type' => self::TYPE_STRING, 'shop' => true, 'validate' => 'isCleanHtml', 'size' => 255),
+            'icone_perso'   => array('type' => self::TYPE_STRING, 'shop' => true, 'validate' => 'isCleanHtml', 'size' => 255),
+            'title'         => array('type' => self::TYPE_STRING, 'shop' => true, 'lang' => true, 'validate' => 'isCleanHtml', 'size' => 255),
+            'description'   => array('type' => self::TYPE_HTML, 'shop' => true, 'lang' => true, 'validate' => 'isCleanHtml', 'size' => 2000),
+            'status'        => array('type' => self::TYPE_BOOL, 'shop' => true, 'validate' => 'isBool', 'required' => true),
+            'position'      => array('type' => self::TYPE_INT,  'shop' => true,'validate' => 'isunsignedInt', 'required' => false),
+            'type_link'     => array('type' => self::TYPE_INT, 'shop' => true, 'validate' => 'isunsignedInt', 'required' => false),
+            'id_cms'     => array('type' => self::TYPE_INT, 'shop' => true, 'validate' => 'isunsignedInt', 'required' => false),
+            'link'          => array('type' => self::TYPE_STRING, 'shop' => true, 'lang' => true, 'validate' => 'isUrl', 'required' => false, 'size' => 255),
+            'date_add' => array('type' => self::TYPE_DATE, 'shop' => true, 'validate' => 'isDate'),
+            'date_upd' => array('type' => self::TYPE_DATE, 'shop' => true, 'validate' => 'isDate'),
+        )
+    );
+
+    /**
+     * getAllBlockByLang
+     *
+     * @param  int $id_lang
+     * @param  int $id_shop
+     *
+     * @return array
+     */
+    public static function getAllBlockByLang($id_lang = 1, $id_shop = 1)
+    {
+        $sql = 'SELECT * FROM `'._DB_PREFIX_.'psreassurance` pr
+            LEFT JOIN '._DB_PREFIX_.'psreassurance_lang prl ON (pr.id_psreassurance = prl.id_psreassurance)
+            WHERE prl.id_lang = "'.(int)$id_lang.'" AND prl.id_shop = "'.(int)$id_shop.'"
+            ORDER BY pr.position';
+
+        $result = Db::getInstance()->executeS($sql);
+
+        return $result;
+    }
+
+    public function handleBlockValues($psr_languages, $type_link)
+    {
+        $languages = Language::getLanguages();
+        $newValues = array();
+
+        foreach ($psr_languages as $key => $value) {
```

Comment:
Not mandatory, but you can use [array_map](https://www.php.net/manual/en/function.array-map.php) to do this

Reply:
keep the information ;) Thank you :+1: